libyear/utils.py

```diff
@@ -9,7 +9,7 @@
 def get_requirement_name_and_version(requirement):
     no_requirement = None, None, None
     # Remove comments if they are on the same line
-    requirement = requirement.split('#')[0].strip()
+    requirement = requirement.split()[0].strip()
```

Comment:
It will fail on requirements which have comment on the same line. 
i.e.
```
requests==2.0.0 # For API requests.
```
Maybe we need both of the lines?

Reply:
with that line it would still work, as there is a space before the `#`, but if there is no space, it wouldn't, so yeah, maybe `.split()[0].split('#')[0]`. I will add a test and fix.

Reply:
A comment before space in requirements file is illegal, so your solution works. Thanks